### PR TITLE
[ENH] add methods section

### LINKED ISSUE #63: automatically generate a method section
- should be easily done with a jinja template
- at first I would just reuse the text that exists in the doc and the docstring of some functions, then we can improve

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes #63 

This is meant as a first rough step.

### CODE REVIEW

File: giga_connectome/data/methods/template.jinja
Comment (Remi-Gau) [inline giga_connectome/data/methods/template.jinja:1]: Probably worth checking with COBIDAS and the bids BEP for connectivity to see what else can go in there.

What are the typical things you would expect to see in a methods section @htwangtw ?

Comment (htwangtw) [reply giga_connectome/data/methods/template.jinja:1]: 1. post processing details done on fMRIPrep output
2. Atlas used 
3. Connectome calculation methods (most common one is pearson's correlation with LedoitWolf covariance estimator)

File: giga_connectome/data/methods/template.jinja [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - Time series extractions through label or map maskers are performed on the denoised nifti file.
+    - Denoising steps were performed on the voxel level:
+        - spatial smoothing (FWHM: {{ data.smoothing_fwhm }} mm)
+        - detrending, only if high pass filter was not applied through confounds
+        - regressing out confounds (using a {{ data.strategy }} strategy)
+        - standardized (using {{ data.standardize }})
+    - Extracted time series from atlas
+    - Computed correlation matrix
+
+5. Calculate intranetwork correlation of each parcel. The values replace the diagonal of the connectomes.
```
Comment (htwangtw) [inline giga_connectome/data/methods/template.jinja:26]: this step is optional; and I feel there's a need to flesh out the details of this thing

Comment (Remi-Gau) [reply giga_connectome/data/methods/template.jinja:26]: OK I will adapt the template to only include it when people asked for it

Comment (Remi-Gau) [reply giga_connectome/data/methods/template.jinja]: this is done: some section of the methods section will only be added if the user passes certain parameters

File: giga_connectome/data/methods/template.jinja
Comment (htwangtw) [reply giga_connectome/data/methods/template.jinja:1]: +1 for waiting on BIDS effort.
What you have here is more than sufficient comparing to what I have seen in papers

**Review by htwangtw — APPROVED** [review]
thank you this is a nice addition

**Pull request merged**